ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/RequestConverter.java

```diff
@@ -178,19 +178,26 @@ private static boolean isResourceIdRequired(final Ds3Request ds3Request) {
     public static String getResponseType(final ImmutableList<Ds3ResponseCode> responseCodes) {
         if (isEmpty(responseCodes)) return "";
 
+        String responseTypeName = "";
         for (final Ds3ResponseCode responseCode : responseCodes) {
             final int rc = responseCode.getCode();
             if (rc < 200 || rc >= 300) continue;
 
             for (final Ds3ResponseType responseType : responseCode.getDs3ResponseTypes()) {
                 if (ConverterUtil.hasContent(responseType.getType()) && !responseType.getType().contentEquals("null")) {
-                    return StructHelper.getResponseTypeName(responseType.getType());
+                    responseTypeName = responseType.getType();
                 }
                 if (ConverterUtil.hasContent(responseType.getComponentType()) && !responseType.getComponentType().contentEquals("null")) {
```

Comment:
Should this be an else if where you first check if the response type has a component type, else check the type?